test/GPv2TradeSimulator.test.ts

```diff
@@ -0,0 +1,233 @@
+        await tokens[1].mock.transfer
+          .withArgs(actualReceiver, buyAmount)
+          .returns(true);
+
+        const { contractBalance, ownerBalance } =
+          await tradeSimulator.simulateTrade(
+            {
+              owner: trader.address,
+              sellToken: tokens[0].address,
+              buyToken: tokens[1].address,
+              receiver,
+              sellAmount,
+              buyAmount,
+            },
+            {
+              [InteractionStage.INTRA]: [
+                {
+                  target: router.address,
+                  callData: router.interface.encodeFunctionData(
+                    "swapExactTokensForTokens",
+                    swapArgs,
+                  ),
+                },
+              ],
+              // Use post-interactions to change mock configuration to simulate
+              // balance changes.
+              [InteractionStage.POST]: [
+                mockReturnsInteraction(
+                  tokens[0],
+                  "balanceOf",
+                  [settlement.address],
+                  [sellRoundingError],
+                ),
+                mockReturnsInteraction(
+                  tokens[1],
+                  "balanceOf",
+                  [settlement.address],
+                  [buyPositiveSlippage],
+                ),
+                mockReturnsInteraction(
+                  tokens[0],
+                  "balanceOf",
+                  [trader.address],
+                  [0],
+                ),
+                mockReturnsInteraction(
+                  tokens[1],
+                  "balanceOf",
+                  [actualReceiver],
+                  [buyAmount],
+                ),
+              ],
+            },
+          );
+
+        expect(contractBalance.sellTokenDelta).to.equal(sellRoundingError);
+        expect(contractBalance.buyTokenDelta).to.equal(buyPositiveSlippage);
+        expect(ownerBalance.sellTokenDelta).to.equal(sellAmount.mul(-1));
+        expect(ownerBalance.buyTokenDelta).to.equal(buyAmount);
+      }
+    });
+
+    it("computes gas usage", async () => {
+      await tokens[0].mock.balanceOf.withArgs(settlement.address).returns(0);
+      await tokens[1].mock.balanceOf.withArgs(settlement.address).returns(0);
+
+      await tokens[0].mock.balanceOf.withArgs(trader.address).returns(0);
+      await tokens[1].mock.balanceOf.withArgs(trader.address).returns(0);
+
+      await tokens[0].mock.transferFrom
+        .withArgs(trader.address, settlement.address, 0)
+        .returns(true);
+      await tokens[1].mock.transfer.withArgs(trader.address, 0).returns(true);
+
+      const { gasUsed } = await tradeSimulator.simulateTrade(
+        {
+          owner: trader.address,
+          sellToken: tokens[0].address,
+          buyToken: tokens[1].address,
+          sellAmount: 0,
+          buyAmount: 0,
+        },
+        {},
+      );
+
+      expect(gasUsed.gt(ethers.constants.Zero)).to.be.true;
```

Comment:
Maybe we could be more specific here, for example testing that the gas used is between 10k and 100k?